sablo/META-INF/resources/sablo/js/websocket.ts

```diff
@@ -260,29 +262,8 @@ webSocketModule.factory('$webSocket',
 				}
 			}
 
-			// data got back from the server
-			if (obj.cmsgid) { // response to event
-				let deferredEvent = deferredEvents[obj.cmsgid];
-				if (deferredEvent != null && angular.isDefined(deferredEvent)) {
-					if (obj.exception) {
-						// something went wrong
-						if (obj[$sabloConverters.TYPES_KEY] && obj[$sabloConverters.TYPES_KEY].exception) {
-							obj.exception = $sabloConverters.convertFromServerToClient(obj.exception, obj[$sabloConverters.TYPES_KEY].exception, undefined, undefined, undefined)
-						}
-						deferredEvent.reject(obj.exception);
-                        $rootScope.$applyAsync(); // not sure that this is needed at all here but it is meant to replace (by making sure a root digest is called) a very old sync $apply that was used here for the reject; it would error out because of that in firefox where an $apply might happen on top of another $apply then in some situations (alerts combined with sync calls to and from server); and that is not allowed by angular 
-					} else {
-						if (obj[$sabloConverters.TYPES_KEY] && obj[$sabloConverters.TYPES_KEY].ret) {
-							obj.ret = $sabloConverters.convertFromServerToClient(obj.ret, obj[$sabloConverters.TYPES_KEY].ret, undefined, undefined, undefined)
-						}
-						deferredEvent.resolve(obj.ret);
-                        $rootScope.$applyAsync(); // not sure that this is needed at all here but it is meant to replace (by making sure a root digest is called) a very old sync $apply that was used here for the resolve; it would error out because of that in firefox where an $apply might happen on top of another $apply then in some situations (alerts combined with sync calls to and from server); and that is not allowed by angular 
-					}
-				} else $log.warn("Response to an unknown handler call dismissed; can happen (normal) if a handler call gets interrupted by a full browser refresh.");
-				delete deferredEvents[obj.cmsgid];
-				$sabloTestability.testEvents();
-				$sabloLoadingIndicator.hideLoading();
-			}
+            // Storing the currentRequestInfo before processing messages, so the message processing code can access it through $webSocket.getCurrentRequestinfo()
+            currentRequestInfo = deferredEvents?.[obj.cmsgid]?.promise?.requestInfo;
```

Comment:
before this line:
    oldRequestInfo = currentRequestInfo;
oldRequestInfo should also be defined above where it is appropriate